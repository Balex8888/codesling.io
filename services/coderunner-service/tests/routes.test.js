import axios from 'axios';
import dotenv from 'dotenv';

dotenv.config();
dotenv.load();

const URL = `${process.env.HOST}:${process.env.PORT}`;

test('Should respond to a {POST /submit-code}', async (done) => {
  let resp = {};
  try {
    resp = await axios.post(`${URL}/submit-code`, {
      code: "// comment",
    });
  } catch (e) {
    console.log('Unexpected error in coderunner-service {POST /submit-code}. e = ', e);
  }
  expect(resp.status).toMatchSnapshot();
  done();
});

test('Should evaluate code correctly and send back stdout', async () => {
  // tuples of [expression, stdout] to test
  const fixtures = [
    "console.log(2)",
    "var x = 'hello world'; console.log(x);",
  ];
  const makeSingleRequest = async (expression) => {
    let resp = {};
    try {
      resp = await axios.post(`${URL}/submit-code`, {
        code: expression,
      });
    } catch (e) {
      console.log(`Unexpected error in coderunner-service {POST /submit-code}.
        expected expression = ${expression}
        resp.data = ${resp.data}
        e = ${e}`
      );
    }
    return resp.data;
  };
  const tests = await Promise.all(fixtures.map(makeSingleRequest));
  tests.forEach(result => expect(result).toMatchSnapshot());
});
