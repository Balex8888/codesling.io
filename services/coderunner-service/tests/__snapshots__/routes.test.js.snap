// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should evaluate code correctly and send back stdout 1`] = `
"2
"
`;

exports[`Should evaluate code correctly and send back stdout 2`] = `
"hello world
"
`;

exports[`Should respond to a {POST /submit-code} 1`] = `200`;
